sonic_platform_base/module_base.py

```diff
@@ -68,6 +86,83 @@ def get_system_eeprom_info(self):
         """
         raise NotImplementedError
 
+    def get_name(self):
+        """
+        Retrieves the name of the module prefixed by SUPERVISOR, LINE-CARD,
+        FABRIC-CARD
+
+        Returns:
+            string: A string providing the name of the card prefixed by one of the
+            MODULE_TYPE_SUPERVISOR, MODULE_TYPE_LINE, MODULE_TYPE_FABRIC followed by
+            an index.
+            Ex. A Chassis having 1 supervisor, 4 line-cards and 6 fabric-cards
+            can provide names SUPERVISOR0, LINE-CARD0 to LINE-CARD3,
+            FABRIC-CARD0 to FABRIC-CARD5
+        """
+        raise NotImplementedError
+
+    def get_description(self):
+        """
+        Retrieves the platform vendor's product description of the module
+
+        Returns:
+            string: A string providing the product description of the module. This
+            is vendor specific.
+        """
+        raise NotImplementedError
+
+    def get_slot(self):
+        """
+        Retrieves the platform vendor's slot number of the module
+
+        Returns:
+            An integer, indicating the slot number in the chassis
+        """
+        raise NotImplementedError
+
+    def get_type(self):
+        """
+        Retrieves the type of the module.
+
+        Returns:
+            string: A string providing the module-type. Supported values are
+            MODULE_TYPE_SUPERVISOR, MODULE_TYPE_LINE, MODULE_TYPE_FABRIC
+        """
+        raise NotImplementedError
+
+    def get_status(self):
```

Comment:
This class inherits from DeviceBase, which already defines a `get_status()` method.

Reply:
Added this mainly to highlight the docstring since it was recommended we need status to be one of the mentioned status-types from MODULE_STATUS_EMPTY, MODULE_STATUS_OFFLINE, MODULE_STATUS_PRESENT, MODULE_STATUS_FAULT and MODULE_STATUS_ONLINE. If not, we can remove it. Please suggest.

Reply:
@mprabhu-nokia: I see you have renamed this to `get_oper_status()`, making it a new, module-specific method. That's good. However, I still wonder if it is necessary. There are already `get_presence()` and `get_status()` methods inherited from DeviceBase, both which return boolean values. `MODULE_STATUS_EMPTY` and `MODULE_STATUS_PRESENT` are redundant with `get_presence()`, and `get_status()` can return True which represents `MODULE_STATUS_ONLINE` or False which would represent either `MODULE_STATUS_OFFLINE` or `MODULE_STATUS_FAULT`. The only thing that might not be easy to differentiate between is `MODULE_STATUS_OFFLINE` or `MODULE_STATUS_FAULT`.

What are your thoughts on this?

Reply:
This was discussed in the review meeting. I had initially proposed just presence and status. However, multiple reviewers wanted more stateful information to be provided and displayed in the DB (and show command). 
Reviewers wanted more states(status strings) and reason-code to be added as well. Those can be taken up as enhancement PRs.